### DISCUSSION TITLE
Can I build an AWS S3 "Object Explorer" with Budibase?

### DISCUSSION THREAD

**Discussion by gitgrooves** [Budibase/budibase#2936] [Help] [answered]
Hello!

The below may be classed as dumb questions so I apologize from the start :)

I have been messing around with Budibase for a few months and notice AWS S3 as a data source. I assume this is a source for pulling data only?

Would there be a way that I can build an S3 "Object Explorer" with Budibase to allow me to access a predefined bucket and then browse through all the available folders?

Could I then upload/overwrite/delete any object/folders? 

I am aware of S3 clients that do all this already but the goal here is to have a simple interface (accessed via OpenID) to allow a user to use the above functionality without having the S3 bucket credentials. 

Now that I wrote all the above I realize this may be completely out of scope for Budibase but still interested to hear what is/isn't possible.

BTW I ❤ Budibase, its constantly expanding my knowledge in how to build an app. Thank you for creating it 🙌

**Comment by mjashanks** [comment, marked as answer]
Hey @gitgrooves 

This is currently not possible.

To be honest, or S3 connector is very basic, and we probably won't be putting effort into it in the near future. It can only be used to pull the top-level contents of a bucket.

It was written a long time ago and doesn't get much use. Happy to accept contributions, if you wish to make it better.

Thanks for the kind words!
